client/apps/edit/components/edit_container.jsx

```diff
@@ -1,39 +1,70 @@
 import PropTypes from 'prop-types'
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
-import { changeSavedStatus, saveArticle } from 'client/actions/editActions'
-
+import {
+  changeSavedStatus,
+  saveArticle,
+  toggleSpinner
+} from 'client/actions/editActions'
 import { ErrorBoundary } from 'client/components/error/error_boundary'
 import { EditAdmin } from './admin/index.jsx'
 import { EditContent } from './content/index.jsx'
 import { EditDisplay } from './display/index.jsx'
 import EditHeader from './header/index.jsx'
 import EditError from './error/index.jsx'
 
-class EditContainer extends Component {
+export class EditContainer extends Component {
   static propTypes = {
     activeView: PropTypes.string,
     article: PropTypes.object,
     changeSavedStatusAction: PropTypes.func,
     channel: PropTypes.object,
     error: PropTypes.object,
+    isSaved: PropTypes.bool,
     saveArticleAction: PropTypes.func,
-    user: PropTypes.object
+    toggleSpinnerAction: PropTypes.func
   }
 
   constructor (props) {
     super(props)
 
-    this.state = {
-      lastUpdated: null
-    }
+    this.beforeUnload = this.beforeUnload.bind(this)
```

Comment:
Unless I'm missing something, you shouldn't need to bind `this` here as `beforeUnload` is defined as bound function using ES6's property initializer (hhttps://babeljs.io/docs/plugins/transform-class-properties/l)

Reply:
Despite using the es6 binding it was not getting the correct 'this' in this instance.